libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -402,6 +402,50 @@ void AP_Mount::handle_gimbal_manager_set_attitude(const mavlink_message_t &msg){
     }
 }
 
+void AP_Mount::handle_command_gimbal_manager_set_pitchyaw(const mavlink_message_t &msg){
+    mavlink_gimbal_manager_set_pitchyaw_t packet;
+    mavlink_msg_gimbal_manager_set_pitchyaw_decode(&msg,&packet);
+
+    // check gimbal device id.  0 is primary, 1 is 1st gimbal, 2 is 2nd gimbal, etc
+    if (packet.gimbal_device_id > AP_MOUNT_MAX_INSTANCES) {
+        return;
+    }
+    const uint8_t gimbal_instance = (packet.gimbal_device_id < 1) ? get_primary_instance() : (uint8_t)packet.gimbal_device_id - 1;
+    if (gimbal_instance >= AP_MOUNT_MAX_INSTANCES || _backends[gimbal_instance] == nullptr) {
+        return;
+    }
+
+    // check flags for change to RETRACT
+    uint32_t flags = (uint32_t)packet.flags;
+    if ((flags & GIMBAL_MANAGER_FLAGS_RETRACT) > 0) {
+        _backends[gimbal_instance]->set_mode(MAV_MOUNT_MODE_RETRACT);
```

Comment:
similar to the above this should be changed to, "backend->set_mode(MAV_MOUNT_MODE_RETRACT);".